docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -1,326 +1,752 @@
++
 [source,java]
 ----
-List<Person> persons = Person.list("order by name,birth");
+PanacheQuery<Person> livingPersons = Person.find("status", Status.Alive);
+----
+** Repository pattern:
++
+[source,java]
+----
+PanacheQuery<PersonRepository> livingPersons = Person.find("status", Status.Alive);
 ----
 
-But these methods also accept an optional `Sort` parameter, which allows your to abstract your sorting:
+* Specify a range from index 0 to index 24 (inclusive):
++
+[source,java]
+----
+livingPersons.range(0, 24);
+----
 
+* Specify a range called `firstRange` with the range index 0 to index 24 (inclusive):
+** Active record pattern:
++
 [source,java]
 ----
-List<Person> persons = Person.list(Sort.by("name").and("birth"));
+List<Person> firstRange = livingPersons.list();
+----
+** Repository pattern:
++
+[source,java]
+----
+List<PersonRepository> firstRange = livingPersons.list();
+----
 
-// and with more restrictions
-List<Person> persons = Person.list("status", Sort.by("name").and("birth"), Status.Alive);
+* Specify a new range called `secondRange` with the range index 25 to index 49 (inclusive):
+** Active record pattern:
++
+[source,java]
+----
+List<Person> secondRange = livingPersons.range(25, 49).list();
 ----
 
-The `Sort` class has plenty of methods for adding columns and specifying sort direction.
+** Repository pattern:
++
+[source,java]
+----
+List<PersonRepository> secondRange = livingPersons.range(25, 49).list();
+----
 
-== Adding entity methods
 
-In general, we recommend not adding custom queries for your entities outside of the entities themselves,
-to keep all model queries close to the models they operate on. So we recommend adding them as static methods
-in your entity class:
+=== Simplified Hibernate ORM with Panache queries
+You can use HQL to write queries for Hibernate ORM with Panache entities, as shown in the following example:
 
 [source,java]
 ----
-@Entity
-public class Person extends PanacheEntity {
-    public String name;
-    public LocalDate birth;
-    public Status status;
+Order.find("select distinct o from Order o left join fetch o.lineItems");
+Order.update("update from Person set name = 'Mortal' where status = ?", Status.Alive);
+----
 
-    public static Person findByName(String name){
-        return find("name", name).firstResult();
-    }
+However, Hibernate ORM with Panache provides several HQL short cuts.
 
-    public static List<Person> findAlive(){
-        return list("status", Status.Alive);
-    }
+.FROM
+In most cases, HQL `SELECT` statements are immediately followed by the `FROM` clause, for example `SELECT name FROM _ENTITY_NAME_`, where `_ENTITY_NAME_` is the name of a Java entity. However, if you do not include `FROM` in your HQL `SELECT` statement, Panache adds `FROM` in the following cases:
 
-    public static void deleteStefs(){
-        delete("name", "Stef");
-    }
-}
-----
+[cols="30%,70%" options="header"]
+
+|===
+h|User entry
+h|Expanded entry
+
+|`ORDER BY`
```

Comment:
I feel like the original `…` which is missing here gave some context that the query had to start with this, and what followed would end up in the resulting HQL translation. Here it looks like the entire query should be just those two tokens, which is wrong.

Reply:
Updated the headings.